app/models/container_node.rb

```diff
@@ -6,6 +6,7 @@ class ContainerNode < ActiveRecord::Base
   belongs_to :ext_management_system, :foreign_key => "ems_id"
   has_many   :container_groups
   has_many   :container_node_conditions, :dependent => :destroy
+  has_many   :labels, -> { where(:section => "labels") }, :class_name => "CustomAttribute", :as => :resource
```

Comment:
> Can't recall don't we need to use :dependent => :destroy here as well? What happens when a node disappears? Are the labels removed as well? (they should).
> 
> Please check that also for other resources the labels are dropped when the entities are deleted.

```ruby :dependent => :destroy ``` is not used on the labels of container entities.
When a node is destroyed the label remains in the custom_attributes table. I'll open an issue regarding that.